packaging/pep517_backend/_backend.py

```diff
@@ -371,10 +372,13 @@ def get_requires_for_build_wheel(
             stacklevel=999,
         )
 
-    c_ext_build_deps = [] if is_pure_python_build else [
-        'Cython ~= 3.0.0; python_version >= "3.12"',
-        'Cython; python_version < "3.12"',
-    ]
+    if is_pure_python_build:
+        c_ext_build_deps = []
+    elif sysconfig.get_config_var("Py_GIL_DISABLED"):
+        # TODO: Remove when there's a Cython final with free-threading support
+        c_ext_build_deps = ['Cython == 3.1.0a1']
```

Comment:
Why not compat?
```suggestion
        c_ext_build_deps = ['Cython ~= 3.1.0a1']
```
It'd allow other versions below 3.1.1

Reply:
That would work for sure. There's still the problem [described here](https://github.com/aio-libs/propcache/pull/84#discussion_r1926906062) however.

Reply:
`3.1.0a2-dev` in Cython master already has a number of thread safety fixes, mostyl minor but including one important one for fused types. Cython 3.1.x is in decent shape now, but `~=` or `>=` seems more useful here.